core/src/main/java/com/netflix/conductor/core/operation/StartWorkflowOperation.java

```diff
@@ -0,0 +1,187 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.core.operation;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.event.EventListener;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+import com.netflix.conductor.core.WorkflowContext;
+import com.netflix.conductor.core.dal.ExecutionDAOFacade;
+import com.netflix.conductor.core.event.WorkflowCreationEvent;
+import com.netflix.conductor.core.event.WorkflowEvaluationEvent;
+import com.netflix.conductor.core.exception.TransientException;
+import com.netflix.conductor.core.execution.StartWorkflowInput;
+import com.netflix.conductor.core.metadata.MetadataMapperService;
+import com.netflix.conductor.core.utils.IDGenerator;
+import com.netflix.conductor.core.utils.ParametersUtils;
+import com.netflix.conductor.metrics.Monitors;
+import com.netflix.conductor.model.WorkflowModel;
+import com.netflix.conductor.service.ExecutionLockService;
+
+@Component
+public class StartWorkflowOperation implements WorkflowOperation<StartWorkflowInput, String> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StartWorkflowOperation.class);
+
+    private final MetadataMapperService metadataMapperService;
+    private final IDGenerator idGenerator;
+    private final ParametersUtils parametersUtils;
+    private final ExecutionDAOFacade executionDAOFacade;
+    private final ExecutionLockService executionLockService;
+    private final ApplicationEventPublisher eventPublisher;
+
+    public StartWorkflowOperation(
+            MetadataMapperService metadataMapperService,
+            IDGenerator idGenerator,
+            ParametersUtils parametersUtils,
+            ExecutionDAOFacade executionDAOFacade,
+            ExecutionLockService executionLockService,
+            ApplicationEventPublisher eventPublisher) {
+        this.metadataMapperService = metadataMapperService;
+        this.idGenerator = idGenerator;
+        this.parametersUtils = parametersUtils;
+        this.executionDAOFacade = executionDAOFacade;
+        this.executionLockService = executionLockService;
+        this.eventPublisher = eventPublisher;
+    }
+
+    @Override
+    @PreAuthorize("hasPermission(#input, 'OWNER')")
+    public String execute(StartWorkflowInput input) {
+        return startWorkflow(input);
+    }
+
+    @EventListener(WorkflowCreationEvent.class)
+    public void handleWorkflowCreationEvent(WorkflowCreationEvent workflowCreationEvent) {
+        startWorkflow(workflowCreationEvent.getStartWorkflowInput());
```

Comment:
This method doesn't have the permission check

Reply:
Good catch. Added the permission check.